jumanji/environments/routing/pacman/utils.py

```diff
@@ -0,0 +1,470 @@
+
+        distance_list = jax.vmap(get_directions, in_axes=(None, 0))(pacman_pos, ghost_p)
+
+        _, ghost_dist = jax.lax.cond(
+            distance_pacman > 8, red_ghost, scared_behaviors, pacman_pos
+        )
+
+        return distance_list, ghost_dist
+
+    def general_behaviors(pacman_pos: Position) -> Tuple[chex.Array, chex.Array]:
+        """
+        General behaviors for ghosts
+        based on https://gameinternals.com/understanding-pac-man-ghost-behavior
+        """
+        blinky = lambda pacman_pos: red_ghost(pacman_pos)
+        inky = lambda pacman_pos: pink_ghost(pacman_pos)
+        pinky = lambda pacman_pos: blue_ghost(pacman_pos)
+        clyde = lambda pacman_pos: orange_ghost(pacman_pos)
+        _, ghost_dist = jax.lax.switch(
+            ghost_num, [blinky, inky, pinky, clyde], pacman_pos
+        )
+        return _, ghost_dist
+
+    def scared_behaviors(pacman_pos: Position) -> Tuple[chex.Array, chex.Array]:
+        """Select scatter targets if power pellet active"""
+        scatter_pos = Position(x=scatter_target[1], y=scatter_target[0])
+        _, ghost_dist = red_ghost(scatter_pos)
+        return _, ghost_dist
+
+    def behaviors() -> Tuple[chex.Array, chex.Array]:
+        """Select scatter or normal targets"""
+        _, ghost_dist = jax.lax.cond(
+            is_scared > 0, scared_behaviors, general_behaviors, pacman_pos
+        )
+        return _, ghost_dist
+
+    def init_behaviors() -> Tuple[chex.Array, chex.Array]:
+        """Set initial fixed behaviors"""
+        target = Position(y=init_target[1], x=init_target[0])
+        _, ghost_dist = red_ghost(pacman_pos=target)
+        return _, ghost_dist
+
+    _, ghost_dist = jax.lax.cond(
+        ghost_init_steps[ghost_num] > 0, init_behaviors, behaviors
+    )
+
+    def get_valid_positions(pos: chex.Array) -> Any:
+        """Get values of surrounding positions"""
+        value = grid[pos[0]][pos[1]]
+        return value
+
+    valids = jax.vmap(get_valid_positions, in_axes=(0,))(ghost_p)
+    valid_locs = valids
+
+    # Set all invalid areas to an infinite cost
+    # remove back tracking
+    valid_no_back = valids * ghost_mask
+    # Get distances of valid locations
+    valid_no_back_d = valid_no_back * ghost_dist
+    invert_mask = valid_no_back != True  # type: ignore # noqa: E712
+    invert_mask = invert_mask * jnp.inf
+    # Set distance of all invalid areas to infinity
+    valid_no_back_d = valid_no_back_d + invert_mask
+    masked_dist = valid_no_back_d
+    minimum_distance = jnp.min(masked_dist)
+    # set all other values aside from the minimum to  0 and the minimum to 1
+    masked_dist = jnp.where(masked_dist == minimum_distance, 1, 0)
+
+    location_value = grid[new_player_pos[1]][new_player_pos[0]]
+    # Check if colliding with walls
+    false_cond = lambda x: False
+    true_cond = lambda x: True
+    collision = jax.lax.cond(location_value == 1, false_cond, true_cond, 0)
```

Comment:
Not sure but this seems equivalent?

```suggestion
    collision = location_value != 1
```